### DISCUSSION TITLE
How would I display this application over a local network?

### DISCUSSION THREAD

**Discussion by HasturDev** [reflex-dev/reflex#1955] [Q&A]
I am attempting to display this application over a local network so I can allow others to interact with the application. Looking at the code it seems that the config is displaying the site over localhost:3000 as a [default](https://github.com/reflex-dev/reflex/blob/891e7ae32dfe9a758ea95534be264b8dfa5b5d79/reflex/config.py#L151). the backend is normally running on 0.0.0.0 and connecting to the api_url which is defaulted above. So do I need to change the backend host Ip address to something other than 0.0.0.0 and instead have it use something like 127.0.0.1? 
I'm probably just missing something obvious as I don't do a lot of front end coding
I'm attempting to get onto it through my phone or other device and I keep getting the error 
"Connection Error"
"Cannot connect to server: websocket error. Check if server is reachable at http://{whatever my site name is set to}:{port}

**Comment by masenf** [comment]
can you post the output of your `reflex run` command with `--loglevel debug` so that I can see all your configuration values.

I just posted a PR today, that should make this a bit more ergonomic for users: https://github.com/reflex-dev/reflex/pull/1954

For now, you have to set the `api_url` config knob to a publicly accessible IP address (or a LAN IP if your testing on the same local network). I'm kind of surprised that you're seeing `{whatever my site name is set to}:{port}` in the websocket error message... that tells me your device is able to access the frontend, but it cannot access the backend. If you go to `http://{whatever my site name is set to}:{port}/ping` do you see the `"pong"` message?

**Reply by HasturDev** [reply]
No I see a 404 message and here are the logs
Also when I change the config file to 
"api_url = "0.0.0.0"" it doesn't give any different logs it's still claiming Debug:   - Local:        http://localhost:3000
[logs.txt](https://github.com/reflex-dev/reflex/files/12887052/logs.txt)

**Reply by HasturDev** [reply]
Here's the logs with the change in api_url just to be sure
Here's the rxconfig.py just in case there's a link that can fix this 
import reflex as rx

config = rx.Config(
    app_name="presentation",
    api_url = "0.0.0.0"
)
[logs(1).txt](https://github.com/reflex-dev/reflex/files/12887067/logs.1.txt)

**Reply by HasturDev** [reply]
I suspect that it could be that I'm attempting to hop onto this on the phone and getting this error. That could be causing the issue. Let me get a second computer to double check

**Reply by masenf** [reply]
You should run with `api_url = "http://192.168.x.y"` (whatever your lan-local IP address is). Where did you see to set `api_url` to `0.0.0.0`?

The `api_url` value has to be accessible by the client browser (frontend).

**Reply by HasturDev** [reply]
0.0.0.0 is usually the wildcard addressv or as the wiki puts it 
`What is 0.0 0.0 used for?
IP address 0.0. 0.0 is used on servers to designate a service may bind to all network interfaces. It tells a server to "listen" for and accept connections from any IP address. On PCs and client devices..` That's what you would use for flask in order to get it to display over the local network. Using the computers Ip address worked, but I'm getting an Unresponsive message on the page. 
[code.txt](https://github.com/reflex-dev/reflex/files/12887168/code.txt)

**Reply by masenf** [reply]
> I'm getting an Unresponsive message on the page.

Is this from the browser?

**Reply by HasturDev** [reply]
> > I'm getting an Unresponsive message on the page.
> 
> Is this from the browser?


yes I tried it from a Safari, Firefox, and Chrome browser

**Reply by masenf** [reply]
ah okay, try with `api_url = "http://192.168.x.y:8000"`, it needs to be an actual URL, i typed too fast in my prior message

**Reply by HasturDev** [reply]
That just gives me json: <html><body>


detail | "Not Found"
-- | --



</body>
</html>

**Reply by masenf** [reply]
The backend listens on port 8000 by default and the frontend is on port 3000 by default. So if you access the 8000 port, it's expected that you get the API response back. When you access on port 3000, you should get the frontend app and it will connect to the backend to establish a websocket connection.
